File: plugins/elasticsearch/src/main/java/com/superior/datatunnel/elasticsearch/ElasticsearchSink.java
package com.superior.datatunnel.elasticsearch;

import com.superior.datatunnel.api.DataTunnelSink;
import com.superior.datatunnel.api.DataTunnelSinkContext;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.elasticsearch.spark.sql.EsSparkSQL;
import scala.collection.JavaConverters;

import java.io.IOException;
import java.util.Map;

public class ElasticsearchSink implements DataTunnelSink<EsSinkOption> {

    @Override
    public void sink(Dataset<Row> dataset, DataTunnelSinkContext<EsSinkOption> context) throws IOException {
        EsSinkOption sinkOption = context.getSinkOption();
        String index = sinkOption.getResource();
        Map<String, String> esCfg = sinkOption.getParams();
        EsSparkSQL.saveToEs(dataset, index, JavaConverters.mapAsScalaMap(esCfg));
    }
}

File: plugins/elasticsearch/src/main/scala/com/superior/datatunnel/elasticsearch/ElasticsearchSink.scala
package com.superior.datatunnel.elasticsearch

import com.superior.datatunnel.api.{DataTunnelSinkContext, DataTunnelSink}
import org.apache.commons.lang3.StringUtils
import org.apache.spark.sql.{Dataset, Row}
import org.elasticsearch.spark.sql._

import scala.collection.JavaConverters._

/**
 * huaixin 2022/2/10 11:40 AM
 */
class ElasticsearchSink extends DataTunnelSink[EsSinkOption] {

  override def sink(dataset: Dataset[Row], context: DataTunnelSinkContext[EsSinkOption]): Unit = {
    val sinkOption = context.getSinkOption
    val index = sinkOption.getResource
    val esCfg = sinkOption.getParams.asScala.filter(item => StringUtils.startsWith(item._1, "es."))
    dataset.saveToEs(index, esCfg)
  }
}
